icu4c/source/i18n/decimfmt.cpp

```diff
@@ -1637,8 +1642,13 @@ void DecimalFormat::touch(UErrorCode& status) {
     setupFastFormat();
 
     // Delete the parsers if they were made previously
+#if U_HAVE_ATOMICS
     delete fields->atomicParser.exchange(nullptr);
     delete fields->atomicCurrencyParser.exchange(nullptr);
+#else
+    delete fields->atomicParser;
```

Comment:
Probably need to assign these two parser fields to be nullptr. The owning object is not being deleted, leaving live invalid pointers.

Reply:
Please just implement the `exchange` function in your shim.